core/src/main/scala/org/apache/spark/metrics/MetricsSystem.scala

```diff
@@ -152,6 +152,12 @@ private[spark] class MetricsSystem private (
     } else { defaultName }
   }
 
+  private[spark] def getRegisteredNames(source: Source): Seq[String] = {
+    val sourceNamePrefix = buildRegistryName(source)
+    val metricNames = source.metricRegistry.getMetrics.keySet()
+    JavaConverters.asScalaSet(metricNames).map(k => s"${sourceNamePrefix}.${k}").toSeq
```

Comment:
1. There is `MetricRegistry.name` method to create names instead of string interpolation.
2. It seems that JavaConverters are not used in such a way [throughout the codebase](https://github.com/apache/spark/search?l=Scala&q=asScala)
3. The result of the method is `ArrayBuffer`, so every lookup is `O(N)`. What is the reason of converting a `Set` into a `Seq`?